include/aspect/material_model/equation_of_state/interface.h

```diff
@@ -113,8 +113,7 @@ namespace aspect
     template <int dim>
     void
     phase_average_equation_of_state_outputs(const EquationOfStateOutputs<dim> &eos_outputs_all_phases,
-                                            const MaterialUtilities::PhaseFunction<dim> &phase_function,
-                                            MaterialUtilities::PhaseFunctionInputs<dim> &phase_in,
+                                            const std::pair<std::vector<double>, const std::vector<unsigned int>> &gamma_inputs,
```

Comment:
You are right that this interface is not ideal, because it would compute the phase function values several times (once for the equation of state, once for the rheology, and so on). But your alternative is somewhat confusing, because `gamma_inputs` does not explain what it is. What about the following interface:

```
    phase_average_equation_of_state_outputs(const EquationOfStateOutputs<dim> &eos_outputs_all_phases,
                                            const std::vector<double> &phase_function_values, 
                                            const std::vector<unsigned int>> &n_phases_per_composition,
                                            EquationOfStateOutputs<dim> &eos_outputs);
```

Please also update the documentation test above this function.